pkg/storage/chunk/client/azure/blob_storage_client.go

```diff
@@ -48,6 +51,11 @@ var (
 		azureUSGovernment: "blob.core.usgovcloudapi.net",
 	}
 
+	defaultAuthFunctions = authFunctions{
+		NewOAuthConfigFunc: adal.NewOAuthConfig,
+		NewServicePrincipalTokenFromFederatedTokenFunc: adal.NewServicePrincipalTokenFromFederatedToken,
```

Comment:
Have you tried to use loki with your WI patch for longer than 24h? I guess it would have the same issue that I discovered while preparing a patch for external-dns and cert-manager. - Federated token is valid for 1h by default, and AAD token received in exchange - for 24h. So, before you can renew an access token, you need to renew the federated token by re-reading it from disk. Unfortunately, adal doesn't expose any native method to do so, but you can trick it through a `customRefreshFunc`. You could find implementation details here: https://github.com/kubernetes-sigs/external-dns/pull/3111
NOTE: it's not like I tested loki with your patch, I'm just assuming you'll hit the same issue.

Reply:
@Tolsto always happy to help :)

Reply:
Can confirm that it now keeps working after 24h.

Reply:
Happy to hear it keeps on working now. The implementation looks fine to me.